### PR TITLE
feat: introducing Qs&As, reddit like tiny platform

### PR DESCRIPTION
**Pull request by siinghd**
### PR feat:
Introducing tiny reddit like platform where people can ask questions or post something and people can reply. 

In the prisma Schema two models are seperated (questions,answers) because with single recursive model it would be hard to get recursive answers.

Screenshot 2024-04-16 at 3 27 23 PM
Screenshot 2024-04-16 at 3 27 37 PM
Screenshot 2024-04-16 at 3 27 52 PM
Screenshot 2024-04-16 at 3 28 09 PM

Resolves #[Issue Number if there] 

### Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I assure there is no similar/duplicate pull request regarding same issue
- [x] I assure that the build i working in local

### CODE REVIEW

**Comment by Neon-20** [comment]
are you thinking off adding a subreddit sort of thing as well, if yes then till which depth. @siinghd.
Something like this also engages community quite much : https://codeforces.com/blog/entry/127903

**Comment by siinghd** [comment]
> are you thinking off adding a subreddit sort of thing as well, if yes then till which depth. @siinghd. Something like this also engages community quite much : https://codeforces.com/blog/entry/127903

This is just like that, looks same as that

**Comment by siinghd** [comment]
@hkirat veere can you look at this?

**Comment by hkirat** [comment]
wow nice
are these attached per video?
or is this a global thing?

**Comment by siinghd** [comment]
@hkirat this is global thing, i think for video we could just expand the comments. We already have different type of comments.

In this somehow we could reference the video... like github does with pr's

**Comment by hkirat** [comment]
Can we put this behind a globally configurable feature flag @siinghd ?
Happy to merge it but need to be able to disable it quickly if there are bugs

**Comment by hkirat** [comment]
Maybe generally add a frontend feature flag functionality if possible

**Comment by siinghd** [comment]
@hkirat sure let me do it

**Comment by siinghd** [comment]
@hkirat  should be done, maybe its dumb but i have added env var NEXT_PUBLIC_DISABLE_FEATURES = "featurea,featureb,featurec"

if you add `qa` or `QA`, basically you disable it, from server side it will redirect the user to '/'. if you are not in the page actions also can't be used i think. let me know if this is what u were thinking

Maybe u also want to disable some things on clientside (like conditional rendering) thats why the env is with NEXT_PUBLIC

**Comment by siinghd** [comment]
MMMM build has some problems with const session = await getServerSession(authOptions), let me check

**Comment by siinghd** [comment]
@hkirat ok veere should have fixed the build, idk why there were issues with session types and other types but should be working now.

But do check one time https://github.com/code100x/cms/pull/436/commits/5e0429ee98c0818991f74baaf84280fe21789491

**Comment by hkirat** [comment]
/bounty $600
Will be testing after merging

File: src/actions/commentVote/index.ts [hunk 1]
```diff
@@ -13,12 +13,18 @@ const voteHandler = async (
   data: InputTypeHandleVote,
 ): Promise<ReturnTypeHandleVote> => {
   const session = await getServerSession(authOptions);
```
Comment (hkirat) [inline src/actions/commentVote/index.ts:15]: ideally should take weather it is a vote or downvote as an input
Toggling can lead to inconsistencies if the user is in two tabs/if the tab hasn't updated

Comment (hkirat) [reply src/actions/commentVote/index.ts:15]: good follow up to add @siinghd

**Pull request merged**